clang/lib/Sema/SemaDecl.cpp

```diff
@@ -20450,9 +20450,12 @@ Sema::DeviceDiagnosticReason Sema::getEmissionReason(const FunctionDecl *FD) {
   if (FD->hasAttr<SYCLSimdAttr>())
     return Sema::DeviceDiagnosticReason::Esimd;
   if (FD->hasAttr<SYCLDeviceAttr>() || FD->hasAttr<SYCLKernelAttr>())
-    return Sema::DeviceDiagnosticReason::Sycl;
+    return getLangOpts().SYCLCUDACompat
+               ? Sema::DeviceDiagnosticReason::SyclCudaCompat
+               : Sema::DeviceDiagnosticReason::Sycl;
   // FIXME: Refine the logic for CUDA and OpenMP.
-  if (getLangOpts().CUDA)
+  // In SYCL-CUDA compat mode, force All just like in normal SYCL
+  if (getLangOpts().CUDA && !getLangOpts().SYCLCUDACompat)
```

Comment:
The commend states, "In SYCL-CUDA compat mode" but the if statement condition checks for `!getLangOpts().SYCLCUDACompat`.

Reply:
Clarify the comment to be less confusing